### PR TITLE
AUTH-1347 - Send the necessary information in SPOT payload

### PR DESCRIPTION
**Pull request by JHjava**
## What?

- Based on what we know now and what is documented in this RFC https://github.com/alphagov/digital-identity-architecture/blob/main/rfc/0013-spot-request-response-structures.md, form the SPOT request payload.

## Why?

- So we can send the relevant information to SPOT

### CODE REVIEW

File: ipv-api/src/main/java/uk/gov/di/authentication/ipv/entity/SPOTClaims.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+public class SPOTClaims {
+
+    @JsonProperty(value = "vot")
+    private String vot;
+
+    @JsonProperty(value = "vtm")
+    private String vtm;
+
+    public SPOTClaims(
```
Comment (chrisclayson) [inline ipv-api/src/main/java/uk/gov/di/authentication/ipv/entity/SPOTClaims.java:13]: Not sure you need this constructor anymore ... if the `ObjectMapper` instance is instantiated with `ObjectMapperFactory.getInstance()` it is configured correctly to just work from the annotation on the private fields.

File: ipv-api/src/main/java/uk/gov/di/authentication/ipv/entity/SPOTRequest.java [hunk 2]
```diff
@@ -4,24 +4,51 @@
...
+    @JsonProperty(value = "in_salt")
+    private byte[] salt;
+
+    @JsonProperty(value = "out_sub")
+    private String sub;
+
+    @JsonProperty(value = "log_ids")
+    private LogIds logIds;
 
     public SPOTRequest(
```
Comment (chrisclayson) [inline ipv-api/src/main/java/uk/gov/di/authentication/ipv/entity/SPOTRequest.java:22]: As above!

File: ipv-api/src/main/java/uk/gov/di/authentication/ipv/entity/SPOTClaims.java (hunk 1 above)
Comment (JHjava) [reply ipv-api/src/main/java/uk/gov/di/authentication/ipv/entity/SPOTClaims.java]: As discussed, keeping the constructor but remove the JsonProperty annotations from it and then adding a default constructor which will be used for desearlizing in the tests.

File: ipv-api/src/main/java/uk/gov/di/authentication/ipv/entity/SPOTRequest.java (hunk 2 above)
Comment (JHjava) [reply ipv-api/src/main/java/uk/gov/di/authentication/ipv/entity/SPOTRequest.java:22]: Address above

**Review by chrisclayson — APPROVED** [review]

**Pull request merged**